src/value.rs

```diff
@@ -263,6 +275,24 @@ impl ValueCommitment {
     pub fn to_bytes(&self) -> [u8; 32] {
         self.0.to_bytes()
     }
+
+    /// x-coordinate of this value commitment.
+    pub fn x(&self) -> pallas::Base {
```

Comment:
This definitely doesn't need to be public:
```suggestion
    pub(in crate::circuit) fn x(&self) -> pallas::Base {
```